OSSM-8740 Document full list of SMCP CR -> Istio CR fields mapping (#88602)
openshift/openshift-docs: modules/ossm-migrating-reference-smcp-configurations.adoc

// Module included in the following assemblies:
//
// * service-mesh-docs-main/migrating/reference/ossm-migrating-reference.adoc

:_mod-docs-content-type: REFERENCE
[id="ossm-migrating-reference-smcp-configurations_{context}"]
= Configuration fields mapping between Service Mesh 2 and Service Mesh 3

> Many of the `ServiceMeshControlPlane` `spec.` configuration fields in {SMProductName} 2 can be configured in the `Istio` resource in {SMProduct} 3.

Reviewer: Having a tough time parsing this:

>  `ServiceMeshControlPlane` `spec.` configuration fields 

Consider alternative phrasings to avoid multiple backticked items in sequence.

The following tables provide guidance for configuring your `Istio` resource in {SMProduct} 3.

> == Cluster configurations

Reviewer: New headings should trigger unique anchor IDs. You could also consider further modularizing this given how many sections there are.

Author: Thanks. That is a post-GA consideration.

Author: Further modularizing, I mean, is a post-GA considerations. Appreciate the reminder on the headings.

| spec.values.global.meshNetworks.gateways.address
| spec.cluster.multiCluster.meshNetworks.gateways.port
| spec.values.global.meshNetworks.gateways.port
| spec.cluster.multiCluster.meshNetworks.gateways.registryServiceName
| spec.values.global.meshNetworks.gateways.registryServiceName
| spec.cluster.name
| spec.values.global.multiCluster.clusterName
| spec.cluster.network
| spec.values.global.network
|===

== General configurations

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.general.logging.componentLevels
| spec.values.global.logging.componentLevels
| spec.general.logging.logAsJSON
| spec.values.global.logAsJson
| spec.general.validationMessages
| spec.values.global.istiod.enableAnalysis
|===

== MeshConfig configurations

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.meshConfig.discoverySelectors
| spec.meshConfig.discoverySelectors
| spec.meshConfig.extensionProviders
| spec.meshConfig.extensionProviders
|===

== Mode configurations

The mode configurations in the {SMProduct} 2 `ServiceMeshControlPlane` resource were:

* Multitenant
* Cluster-wide
* Federation

> 

Reviewer 2: Little misunderstanding here. We still need the `In {SMProduct} 3, the mode is not configured using a single field in the `Istio` resource. By default the {SMProduct}3 control plane has access to all namespaces which is an equivalent for cluster-wide mode in {SMProduct} 2. A similar configuration to multitenant mode in {SMProduct} 2 can be achieved by configuring the `discoverySelectors` field. See "Deploying multiple service meshes on a single cluster".` Paragraph here, it can be standard paragraph not a note.

> In {SMProduct} 3, the mode is not configured using a single field in the `Istio` resource.

Reviewer: ```suggestion
In {SMProduct} 3, the mode is not configured by using a single field in the `Istio` resource.
```

> By default, the {SMProduct} 3 control plane has access to all namespaces which is equivalent to cluster-wide mode in {SMProduct} 2. For a similar configuration to multitenant mode in {SMProduct} 2 in {SMProduct} 3, you need to use the `discoverySelectors` field. See "Deploying multiple service meshes on a single cluster".

Reviewer: ```suggestion
By default, the {SMProduct} 3 control plane has access to all namespaces, which is equivalent to cluster-wide mode in {SMProduct} 2. For a similar configuration to multitenant mode in {SMProduct} 2 in {SMProduct} 3, you must use the `discoverySelectors` field. For more information, see "Deploying multiple service meshes on a single cluster".
```

Also, "multitenant" or "multi-tenant"?

== Profile configurations

The profile configuration options for {SMProduct} 3 are:

> * ambient

Reviewer: I'm not precisely sure what these are given my lack of familiarity with SMCP, but they might need to be in monospace: https://www.ibm.com/docs/en/ibm-style?topic=format-highlighting

| spec.runtime.components.pod.affinity.podAntiAffinity
| spec.values.pilot.affinity.podAntiAffinity
| spec.runtime.components.pod.metadata.annotations
| spec.values.pilot.podAnnotations
| spec.runtime.components.pod.metadata.labels
| spec.values.pilot.podLabels
| spec.runtime.components.pod.nodeSelector
| spec.values.pilot.nodeSelector
| spec.runtime.components.pod.tolerations
| spec.values.pilot.tolerations
|===

=== Defaults configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.runtime.defaults.container.imagePullPolicy
| spec.values.global.imagePullPolicy
| spec.runtime.defaults.container.imagePullSecrets
| spec.values.global.imagePullSecrets
| spec.runtime.defaults.container.imageRegistry
| spec.values.global.hub
| spec.runtime.defaults.container.imageTag
| spec.values.global.tag
| spec.runtime.defaults.container.resources
| spec.values.global.defaultResources
| spec.runtime.defaults.deployment.podDisruption.enabled
| spec.values.global.defaultPodDisruptionBudget.enabled
| spec.runtime.defaults.pod.nodeSelector
| spec.values.global.defaultNodeSelector
| spec.runtime.defaults.pod.priorityClassName

> | spec.values.global.priorityClassName

Reviewer 3: Please remove this "spec.values.global.priorityClassName"  value, it is deprecated.

| spec.cluster.multiCluster.enabled
| spec.values.global.multiCluster.enabled
| spec.cluster.multiCluster.meshNetworks
| spec.values.global.meshNetworks
| spec.cluster.multiCluster.meshNetworks.endpoints
| spec.values.global.meshNetworks.endpoints
| spec.cluster.multiCluster.meshNetworks.endpoints.fromCID
| spec.values.global.meshNetworks.endpoints.fromCidr
| spec.cluster.multiCluster.meshNetworks.endpoints.fromRegistry
| spec.values.global.meshNetworks.endpoints.fromRegistry
| spec.cluster.multiCluster.meshNetworks.gateways
| spec.values.global.meshNetworks.gateways
| spec.cluster.multiCluster.meshNetworks.gateways.address
| spec.values.global.meshNetworks.gateways.address
| spec.cluster.multiCluster.meshNetworks.gateways.port
| spec.values.global.meshNetworks.gateways.port
| spec.cluster.multiCluster.meshNetworks.gateways.registryServiceName
| spec.values.global.meshNetworks.gateways.registryServiceName
| spec.cluster.name
| spec.values.global.multiCluster.clusterName
| spec.cluster.network
| spec.values.global.network
|===

== General configurations

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.general.logging.componentLevels

> | spec.values.global.logging.componentLevels

Reviewer 3: spec.values.global.logging.levels

![Screenshot From 2025-02-25 13-08-31](https://github.com/user-attachments/assets/5372ab06-fe84-4199-9413-01b83b748ed3)

| spec.general.logging.logAsJSON
| spec.values.global.logAsJson
| spec.general.validationMessages
| spec.values.global.istiod.enableAnalysis
|===

== MeshConfig configurations

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.meshConfig.discoverySelectors

> | spec.meshConfig.discoverySelectors

Reviewer 3: spec.values.meshConfig.discoverySelectors

![Screenshot From 2025-02-25 13-19-08](https://github.com/user-attachments/assets/cc72e4a3-e76f-43b4-bd8c-b2c9db82c31f)

| spec.meshConfig.extensionProviders

> | spec.meshConfig.extensionProviders

Reviewer 3: spec.values.meshConfig.extensionProviders

|===

== Mode configurations

The mode configurations in the {SMProduct} 2 `ServiceMeshControlPlane` resource were:

* Multitenant
* Cluster-wide
* Federation

In {SMProduct} 3, the mode is not configured using a single field in the `Istio` resource.

> By default the {SMProduct}3 control plane has access to all namespaces which is equivalent to cluster-wide mode in {SMProduct} 2. For a similar configuration to multitenant mode in {SMProduct} 2 in {SMProduct 3}, you need to use the `discoverySelectors` field. See "Deploying multiple service meshes on a single cluster"

Reviewer 2: ```suggestion
By default the {SMProduct} 3 control plane has access to all namespaces which is equivalent to cluster-wide mode in {SMProduct} 2. For a similar configuration to multitenant mode in {SMProduct} 2 in {SMProduct} 3, you need to use the `discoverySelectors` field. See "Deploying multiple service meshes on a single cluster"
```

> By default, the {SMProduct}3 control plane has access to all namespaces which is equivalent to cluster-wide mode in {SMProduct} 2. For a similar configuration to multitenant mode in {SMProduct} 2 in {SMProduct 3}, you need to use the `discoverySelectors` field. See "Deploying multiple service meshes on a single cluster"

Reviewer 2: ```suggestion
By default, the {SMProduct} 3 control plane has access to all namespaces which is equivalent to cluster-wide mode in {SMProduct} 2. For a similar configuration to multitenant mode in {SMProduct} 2 in {SMProduct} 3, you need to use the `discoverySelectors` field. See "Deploying multiple service meshes on a single cluster"
```

== Profile configurations

The profile configuration options for {SMProduct} 3 are:

* ambient
* default
* demo
* empty
* openshift-ambient
* openshift
* preview
* stable

== Proxy configurations

=== Access logging configuration fields

[cols="1,1"]
|===

> | SMCP 2.6 configuration | Istio 3.0 configuration

Reviewer 3: for all the below 3,0

Add the "  spec.values.meshConfig.<existing_info>

| spec.proxy.accessLogging.envoyService.address
| spec.meshConfig.defaultConfig.envoyAccessLogService.address
| spec.proxy.accessLogging.envoyService.enabled
| spec.meshConfig.enableEnvoyAccessLogService
| spec.proxy.accessLogging.envoyService.tcpKeepalive
| spec.meshConfig.defaultConfig.envoyAccessLogService.tcpKeepalive
| spec.proxy.accessLogging.envoyService.tcpKeepalive.interval
| spec.meshConfig.defaultConfig.envoyAccessLogService.tcpKeepalive.interval
| spec.proxy.accessLogging.envoyService.tcpKeepalive.probes
| spec.meshConfig.defaultConfig.envoyAccessLogService.tcpKeepalive.probes
| spec.proxy.accessLogging.envoyService.tcpKeepalive.time
| spec.meshConfig.defaultConfig.envoyAccessLogService.tcpKeepalive.time
| spec.proxy.accessLogging.envoyService.tlsSettings
| spec.meshConfig.defaultConfig.envoyAccessLogService.tlsSettings
| spec.proxy.accessLogging.file.encoding
| spec.meshConfig.accessLogEncoding
| spec.proxy.accessLogging.file.format
| spec.meshConfig.accessLogFormat
| spec.proxy.accessLogging.file.name
| spec.meshConfig.accessLogFile
|===

=== Basic proxy configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.proxy.adminPort

> | spec.meshConfig.defaultConfig.proxyAdminPort

Reviewer 3: spec.values.meshConfig.defaultConfig.proxyAdminPort

| spec.proxy.concurrency

> | spec.meshConfig.defaultConfig.concurrency

Reviewer 3: spec.values.meshConfig.defaultConfig.concurrency

|===

=== Envoy metrics service fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

> 

Reviewer 3: spec.values.meshConfig.<Existing_value>

| spec.proxy.envoyMetricsService.address
| spec.meshConfig.defaultConfig.envoyMetricsService.address
| spec.proxy.envoyMetricsService.enabled
| spec.meshConfig.enableEnvoyAccessLogService
| spec.proxy.envoyMetricsService.tcpKeepalive
| spec.meshConfig.defaultConfig.envoyMetricsService.tcpKeepalive
| spec.proxy.envoyMetricsService.tlsSettings
| spec.meshConfig.defaultConfig.envoyMetricsService.tlsSettings
|===

=== Injection configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.proxy.injection.alwaysInjectSelector

> | spec.sidecarInjectorWebhook.alwaysInjectSelector

Reviewer 3: spec.values.sidecarInjectorWebhook.<existing_value>

| spec.proxy.injection.neverInjectSelector

> | spec.sidecarInjectorWebhook.neverInjectSelector

Reviewer 3: spec.values.sidecarInjectorWebhook.<existing_value>

| spec.proxy.injection.injectedAnnotations

> | spec.sidecarInjectorWebhook.injectedAnnotations

Reviewer 3: spec.values.sidecarInjectorWebhook.<existing_value>

| spec.proxy.injection.autoInject
| spec.values.global.proxy.autoInject
|===

=== Proxy logging configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.proxy.logging.componentLevels
| spec.values.global.proxy.componentLogLevel
| spec.proxy.logging.level
| spec.values.global.logging.level
|===

=== Proxy networking configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.proxy.networking.clusterDomain
| spec.values.global.proxy.clusterDomain
| spec.proxy.networking.connectionTimeout

> | spec.meshConfig.connectTimeout

Reviewer 3: spec.values.meshConfig.<existing_value>

| spec.proxy.networking.dns.refreshRate

> | spec.meshConfig.dnsRefreshRate

Reviewer 3: spec.values.meshConfig.<existing_value>

| spec.proxy.networking.dns.searchSuffixes
| spec.values.global.podDNSSearchNamespaces
| spec.proxy.networking.initialization.initContainer.runtime.imageName
| spec.values.global.proxy_init.image
| spec.proxy.networking.initialization.initContainer.runtime.imagePullPolicy
| spec.values.global.imagePullPolicy
| spec.proxy.networking.initialization.initContainer.runtime.imagePullSecrets
| spec.values.global.imagePullSecrets
| spec.proxy.networking.initialization.initContainer.runtime.imageRegistry
| spec.values.global.hub
| spec.proxy.networking.initialization.initContainer.runtime.imageTag
| spec.values.global.tag
| spec.proxy.networking.initialization.initContainer.runtime.resources
| spec.values.global.proxy_init.resources
| spec.proxy.networking.maxConnectionAge
| spec.values.pilot.keepaliveMaxServerConnectionAge
| spec.proxy.networking.protocol.timeout

> | spec.meshConfig.protocolDetectionTimeout

Reviewer 3: spec.values.meshConfig.<existing_value>

|===

=== Traffic control configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.proxy.networking.trafficControl.inbound.excludedPorts
| spec.values.global.proxy.excludeInboundPorts
| spec.proxy.networking.trafficControl.inbound.includedPorts
| spec.values.global.proxy.includeInboundPorts
| spec.proxy.networking.trafficControl.inbound.interceptionMode
| spec.values.meshConfig.defaultConfig.interceptionMode
| spec.proxy.networking.trafficControl.outbound.excludedIPRanges
| spec.values.global.proxy.excludeIPRanges
| spec.proxy.networking.trafficControl.outbound.excludedPorts
| spec.values.global.proxy.excludeOutboundPorts
| spec.proxy.networking.trafficControl.outbound.includedIPRanges
| spec.values.global.proxy.includeIPRanges
| spec.proxy.networking.trafficControl.outbound.policy

> | spec.meshConfig.outboundTrafficPolicy.mode

Reviewer 3: spec.values.meshConfig.outboundTrafficPolicy.mode

|===

=== Proxy runtime configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.proxy.runtime.container.env

> | spec.values.global.proxy.env

Reviewer 3: | spec.values.global.proxy.env

I don't this value, will check with the team

Reviewer 3: spec.values.meshConfig.defaultConfig.proxyMetadata

| spec.runtime.components.deployment.autoScaling.targetCPUUtilizationPercentage
| spec.values.pilot.cpu.targetAverageUtilization
| spec.runtime.components.deployment.replicas
| spec.values.pilot.replicaCount
| spec.runtime.components.deployment.strategy.rollingUpdate.maxSurge
| spec.values.pilot.rollingMaxSurge
| spec.runtime.components.deployment.strategy.rollingUpdate.maxUnavailable
| spec.values.pilot.rollingMaxUnavailable
|===

=== Pod configuration fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.runtime.components.pod.affinity
| spec.values.pilot.affinity
| spec.runtime.components.pod.affinity.nodeAffinity
| spec.values.pilot.affinity.nodeAffinity
| spec.runtime.components.pod.affinity.podAffinity
| spec.values.pilot.affinity.podAffinity
| spec.runtime.components.pod.affinity.podAntiAffinity
| spec.values.pilot.affinity.podAntiAffinity
| spec.runtime.components.pod.metadata.annotations
| spec.values.pilot.podAnnotations
| spec.runtime.components.pod.metadata.labels
| spec.values.pilot.podLabels
| spec.runtime.components.pod.nodeSelector
| spec.values.pilot.nodeSelector
| spec.runtime.components.pod.priorityClassName

> | spec.values.pilot.priorityClassName

Reviewer 3: Thia value is not available, checking with the team.

Reviewer 3: | spec.values.pilot.priorityClassName
It is deprecated remove from the doc.

| spec.security.certificateAuthority.cert-manager.address
| spec.values.meshConfig.ca.address
| spec.security.certificateAuthority.custom.address
| spec.values.meshConfig.ca.address
|===

=== Istiod CA fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.security.certificateAuthority.istiod.type
| spec.values.global.pilotCertProvider
|===

=== Control plane security fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.security.controlPlane.certProvider
| spec.values.global.pilotCertProvider
| spec.security.controlPlane.mtls
| spec.values.meshConfig.enableAutoMtls
| spec.security.controlPlane.tls.cipherSuites
| spec.values.meshConfig.tlsDefaults.cipherSuites
| spec.security.controlPlane.tls.ecdhCurves
| spec.values.meshConfig.tlsDefaults.ecdhCurves
| spec.security.controlPlane.tls.minProtocolVersion
| spec.values.meshConfig.tlsDefaults.minProtocolVersion
|===

=== Data plane security fields

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.security.dataPlane.automtls
| spec.values.meshConfig.enableAutoMtls
| spec.security.dataPlane.mtls

> | spec.values.meshConfig.meshMTLS.enabled

Reviewer 3: spec.values.meshConfig.meshMTLS

remove the "enabled" from the above one

// Module included in the following assemblies:
//
// * service-mesh-docs-main/migrating/reference/ossm-migrating-reference.adoc

:_mod-docs-content-type: REFERENCE
[id="ossm-migrating-reference-smcp-configurations_{context}"]

> = Configuration fields between Service Mesh 2 and Service Mesh 3

Reviewer 2: ```suggestion
= Configuration fields mapping between Service Mesh 2 and Service Mesh 3
```

| spec.values.global.meshNetworks.gateways.address
| spec.cluster.multiCluster.meshNetworks.gateways.port
| spec.values.global.meshNetworks.gateways.port
| spec.cluster.multiCluster.meshNetworks.gateways.registryServiceName
| spec.values.global.meshNetworks.gateways.registryServiceName
| spec.cluster.name
| spec.values.global.multiCluster.clusterName
| spec.cluster.network
| spec.values.global.network
|===

== General configurations

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.general.logging.componentLevels
| spec.values.global.logging.componentLevels
| spec.general.logging.logAsJSON
| spec.values.global.logAsJson
| spec.general.validationMessages
| spec.values.global.istiod.enableAnalysis
|===

== MeshConfig configurations

[cols="1,1"]
|===
| SMCP 2.6 configuration | Istio 3.0 configuration

| spec.meshConfig.discoverySelectors
| spec.meshConfig.discoverySelectors
| spec.meshConfig.extensionProviders
| spec.meshConfig.extensionProviders
|===

== Mode configurations

The mode configurations in the {SMProduct} 2 `ServiceMeshControlPlane` resource were:

* Multitenant
* Cluster-wide
* Federation

> The mode configuration options in {SMProduct} 3 are:
> 
> * Cluster-wide (default)
> * Multitenant
> 
> [NOTE]
> ====

Reviewer 2: ```suggestion
```

> The mode configuration options in {SMProduct} 3 are:
> 
> * Cluster-wide (default)
> * Multitenant
> 
> [NOTE]
> ====
> in {SMProduct}3, the mode is not configured using a single field in the `Istio` resource. To configure multitenant mode, you also need to configure the `discoverySelectors` field.
> ====

Reviewer 2: ```suggestion
In {SMProduct} 3, the mode is not configured using a single field in the `Istio` resource. By default the {SMProduct}3 control plane will have access to all namespaces which is an equivalent for cluster-wide mode in {SMProduct} 2. Similar configuration to multitenant mode known in {SMProduct} 2 can be achieved by configuring the `discoverySelectors`. See `Deploying multiple service meshes on a single cluster` for details.
```

> ====

Reviewer 2: ```suggestion
```
